tests/jlm/util/TestBiMap.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2023 Håvard Krogstie <krogstie.havard@gmail.com>
+ * See COPYING for terms of redistribution.
+ */
+
+#include <test-registry.hpp>
+
+#include <jlm/util/BiMap.hpp>
+
+#include <cassert>
+#include <string>
+
+#define SHOULD_THROW_ERROR(code)                       \
+  do                                                   \
+  {                                                    \
+    try                                                \
+    {                                                  \
+      code;                                            \
+      assert(false && #code " was supposed to throw"); \
+    }                                                  \
+    catch (jlm::util::error e)                         \
+    {                                                  \
+      (void)e;                                         \
+    }                                                  \
+  } while (false)
```

Comment:
We should put this somewhere else. This is a very useful code snippet. There are other places in the unit tests where this piece would be useful.